src/Workspaces/Remote/ServiceHub/Host/AssetProvider.cs

```diff
@@ -39,24 +39,24 @@ internal sealed partial class AssetProvider(Checksum solutionChecksum, SolutionA
         using var _1 = PooledHashSet<Checksum>.GetInstance(out var checksums);
         checksums.Add(checksum);
 
-        using var _2 = PooledDictionary<Checksum, object>.GetInstance(out var results);
+        using var _2 = PooledDictionary<Checksum, T>.GetInstance(out var results);
         await this.SynchronizeAssetsAsync(assetPath, checksums, results, cancellationToken).ConfigureAwait(false);
 
-        return (T)results[checksum];
+        return results[checksum];
     }
 
     public override async ValueTask<ImmutableArray<(Checksum checksum, T asset)>> GetAssetsAsync<T>(
         AssetPath assetPath, HashSet<Checksum> checksums, CancellationToken cancellationToken)
     {
-        using var _ = PooledDictionary<Checksum, object>.GetInstance(out var results);
+        using var _ = PooledDictionary<Checksum, T>.GetInstance(out var results);
 
         await this.SynchronizeAssetsAsync(assetPath, checksums, results, cancellationToken).ConfigureAwait(false);
```

Comment:
the method this is in is already generic.  so this now allows that generic information to flow *into* SynchronizeAssetsAsync